### PR TITLE
[Features] Add features for fade in out hero unit background

### PR DESCRIPTION
**Pull request by ratchagarn**
**1. Objective reason**

Because `Reactbkk team` lazy to do this features by themselves.
Related issue [#1](https://github.com/reactbkk/2.0.0/issues/1) 

**2. Description of change**

- Add new hero background images.
- Add css for handling animation and center crop images.
- Add component for handling hero unit background fade in/out.

### CODE REVIEW

File: components/HeroUnit.js [hunk 1]
```diff
@@ -84,3 +86,67 @@ function HeroUnitFooter () {
     </div>
   )
 }
+
+class HeroUnitBg extends React.Component {
+  componentDidMount () {
+    const { delay } = this.props
+    const bgHeroSet = ReactDOM.findDOMNode(this)
```
Comment (dtinth) [inline components/HeroUnit.js:93]: I recommend using React’s `ref` instead of using `ReactDOM.findDOMNode`.

File: components/HeroUnit.js [hunk 2]
```diff
@@ -84,3 +86,67 @@ function HeroUnitFooter () {
...
+    const img = bgHeroSet.querySelectorAll('img')
+
+    let pointer = 0
+    const maxItems = img.length
+
+    setInterval(() => {
```
Comment (dtinth) [inline components/HeroUnit.js:99]: When using `setInterval` in `componentDidMount`, it’s best to save a reference to the interval ID, and do a `clearInterval` in `componentWillUnmount`.

File: components/HeroUnit.js [hunk 3]
```diff
@@ -84,3 +86,67 @@ function HeroUnitFooter () {
...
+      if (pointer < maxItems - 1) {
+        img[pointer].className = 'fade-out'
+        img[pointer + 1].className = 'fade-in'
+        pointer++
+      } else {
+        img[pointer].className = 'fade-out'
+        img[0].className = 'fade-in'
+        pointer = 0
```
Comment (dtinth) [inline components/HeroUnit.js:107]: I think this code can be simplified to use modulo arithmetic. For example:

```js
const nextPointer = (pointer + 1) % img.length
img[pointer].className = 'fade-out'
img[nextPointer].className = 'fade-in'
pointer = nextPointer
```

**Comment by dtinth** [comment]
Hello,

Thanks again for the contribution! I’ve reviewed your code.

However, the approach I recommend is using CSS-only animations, because it can be run when the page has not finished loading. Also, you already got a ticket, so I will give this issue to #49 (where we are discussing about technology choices).

Thank you for understanding, and thanks again for your help. Much appreciated!

**Comment by ratchagarn** [comment]
@dtinth I understand your reason. It was fun for me.
Thanks for your suggestion. 🙇

**Pull request closed without merging**